### PR TITLE
 add clap command to show secure-channel listener

### LINKED ISSUE #4439: Create clap command to show the details of an existing secure-channel listener on a node
### Current behavior

Missing command for `secure-channel-listener show`.

### Desired behavior

NOTE: currently, we only store the address of the secure-channel listener, so this command is rather useless as the input is the same as the information it outputs. In the near future though, we want to modify the information that is shown and could include more data about the listener.

Implement a command that retrieves from a running node the information of a `secure-channel-listener` given its address.

```bash
ockam secure-channel-listener show [address] --at [node-name]
```

Required args:
- address

Optional args:
- at: defaults to the default node

Output: same as described at [`secure-channel-listener create`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/secure_channel/listener/create.rs#L73).

### PR DESCRIPTION
**Pull request by murex971**
Resolves #4439

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/nodes/models/secure_channel.rs [hunk 1]
```diff
@@ -165,6 +165,46 @@ impl<'a> DeleteSecureChannelListenerResponse<'a> {
     }
 }
 
+/// Request body to show a Secure Channel Listener
+#[derive(Debug, Clone, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct ShowSecureChannelListenerRequest<'a> {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<8293631>,
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/nodes/models/secure_channel.rs]: the cbor encoding is perfect! but the random number of the `TypeTag` is already taken by another structure. Could you please change the number with a new random number?

You can use this generator: https://www.random.org/integers/?num=100&min=1000000&max=9999999&col=5&base=10&format=html&rnd=new

File: implementations/rust/ockam/ockam_api/src/nodes/models/secure_channel.rs [hunk 2]
```diff
@@ -165,6 +165,46 @@ impl<'a> DeleteSecureChannelListenerResponse<'a> {
...
+    }
+}
+
+/// Response body to show a Secure Channel Listener
+#[derive(Debug, Clone, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct ShowSecureChannelListenerResponse<'a> {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<8642885>,
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/nodes/models/secure_channel.rs]: Same as before, everything looks good, only the `TypeTag` random needs an update.

File: implementations/rust/ockam/ockam_command/src/secure_channel/listener/show.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    let node = extract_address_value(at)?;
+    let address = &cmd.address;
+
+    let mut rpc = Rpc::background(ctx, &opts, &node)?;
+    let req = api::show_secure_channel_listener(address);
+    rpc.request(req).await?;
+    rpc.is_ok()?;
+
+    println!(
+        "secure-channel listener with address '/service/{}'",
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_command/src/secure_channel/listener/show.rs]: Could you please simplify the output to align it to the creation command?

https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/secure_channel/listener/create.rs#L73

File: implementations/rust/ockam/ockam_command/src/secure_channel/listener/mod.rs [hunk 4]
```diff
@@ -31,6 +33,8 @@ pub enum SecureChannelListenerSubcommand {
     Delete(DeleteCommand),
     #[command(display_order = 800)]
     List(ListCommand),
+    #[command(display_order = 800)]
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_command/src/secure_channel/listener/mod.rs]: Seems like every command wants to be displayed with the same priority! :smile: 
`display_order` should be incremental, smaller numbers are displayed before others.
Would you do me the favor to update also previously existing `display_order` from 800 to 803?

**Review by davide-baldo — CHANGES_REQUESTED** [review]
Thanks for your contribution!
it looks fantastic :+1:  only minor changes are required!

**Comment by murex971** [comment]
@davide-baldo  thanks for the review. 
Made all the changes :)

**Review by davide-baldo — DISMISSED** [review]
nice! :+1: 
I can't merge right away due to an upcoming release, but we'll merge as soon as we can!

**Review by etorreborre — DISMISSED** [review]

**Review by SanjoDeundiak — APPROVED** [review]

**Pull request merged**